src/Agent.Worker/TestResults/TrxResultReader.cs

```diff
@@ -389,6 +391,35 @@ private List<TestCaseResultData> ReadActualResults(XmlNodeList resultsNodes, str
 
                 resultCreateModel.RunBy = _runUserIdRef;
 
+                List<string> resulLeveltAttachments = new List<string>() { };
+
+                XmlNodeList resultAttachmentNodes = resultNode.SelectNodes("CollectorDataEntries/Collector/UriAttachments/UriAttachment/A");
+                if (resultAttachmentNodes.Count > 0 && executionId != null)
+                {
+                    foreach (XmlNode resultAttachmentNode in resultAttachmentNodes)
+                    {
+                        if (resultAttachmentNode.Attributes["href"] != null && resultAttachmentNode.Attributes["href"].Value != null)
+                        {
+                            resulLeveltAttachments.Add(Path.Combine(_attachmentLocation, executionId, resultAttachmentNode.Attributes["href"].Value));
+                        }
+                    }
+                }
+
+                XmlNodeList resultFileNodes = resultNode.SelectNodes("ResultFiles/ResultFile");
+
+                if (resultFileNodes.Count > 0 && executionId != null)
+                {
+                    foreach (XmlNode resultFileNode in resultFileNodes)
+                    {
+                        if (resultFileNode.Attributes["path"] != null && resultFileNode.Attributes["path"].Value != null)
+                        {
+                            resulLeveltAttachments.Add(Path.Combine(_attachmentLocation, executionId, resultFileNode.Attributes["path"].Value));
+                        }
+                    }
+                }
+
+                resultCreateModel.AttachmentData = new AttachmentData() { AttachmentsFilePathList = resulLeveltAttachments.ToArray() };
```

Comment:
Refactor into a new method

Reply:
TestCaseResult and TestCaseSubResult are 2 different objects.

Reply:
That's okay, method looks very big. So wanted to break it `GetResultLevelAttachments`, `GetInnerResults` which makes this more readable.